File: examples/sbt/src/main/java/com/twitter/jexample/EchoLoadTest.java
package com.twitter.jexample;

import com.twitter.example.thrift.EchoService;
import com.twitter.parrot.processor.ThriftLoadTest;
import com.twitter.parrot.server.ParrotRequest;
import com.twitter.parrot.server.ParrotService;
import com.twitter.parrot.thrift.ParrotJob;
import com.twitter.util.Future;
import com.twitter.util.FutureEventListener;
import org.apache.thrift.protocol.TBinaryProtocol;

import java.util.List;

public class EchoLoadTest extends ThriftLoadTest {
    EchoService.ServiceToClient client = null;

    public EchoLoadTest(ParrotService<ParrotRequest, byte[]> parrotService) {
	super(parrotService);
        client = new EchoService.ServiceToClient(service(), new TBinaryProtocol.Factory());
    }

    public void processLines(ParrotJob job, List<String> lines) {
        for(String line: lines) {
            Future<String> future = client.echo(line);
            future.addEventListener(new FutureEventListener<String>() {
                public void onSuccess(String msg) {
                    System.out.println("response: " + msg);
                }

                public void onFailure(Throwable cause) {
                    System.out.println("Error: " + cause);
                }
            });
        }
    }
}

File: examples/echo/src/main/scala/com/twitter/example/EchoLoadTest.scala
package com.twitter.example

import org.apache.thrift.protocol.TBinaryProtocol

import com.twitter.parrot.processor.ThriftRecordProcessor
import com.twitter.parrot.server.{ParrotRequest, ParrotService}
import com.twitter.logging.Logger

import thrift.EchoService

class EchoLoadTest(parrotService: ParrotService[ParrotRequest, Array[Byte]]) extends ThriftRecordProcessor(parrotService) {
  val client = new EchoService.ServiceToClient(service, new TBinaryProtocol.Factory())
  val log = Logger.get(getClass)

  def processLines(lines: Seq[String]) {
    lines map {
      line =>
        client.echo(line) respond {
          rep =>
            if (rep == "hello") {
              client.echo("OMIGOD IT'S TALKING TO US")
            }
            log.info("response: " + rep)
        }
    }
  }
}
